src/FilenameParsing/HashFileIDHelper.php

```diff
@@ -73,6 +73,9 @@ public function buildFileID($filename, $hash = null, $variant = null, $cleanfile
 
     public function cleanFilename($filename)
     {
+        // Swap backslash for forward slash
+        $filename = str_replace('\\', '/', $filename);
```

Comment:
does this have any impact running on windows?

Reply:
No we're not meant to be using \ even on windows. When you actually end up writing files, flysystem will swap backslashes for forward slashes. However a lot of our own logic won't.

This is why the current logic doesn't fail outright, but places the file in an unexpected location.

Reply:
ok sounds good